scripts/generate-utility-data.ts

```diff
@@ -256,21 +262,45 @@ enum Col {
     }
     seen.add(row.Zip + id);
 
-    names.set(id, name);
+    if (!utilitiesByState.has(row.State)) {
+      utilitiesByState.set(row.State, new Map());
+    }
+    utilitiesByState.get(row.State)!.set(id, { name });
+
     zipToUtilityOut.write({
       zip: row.Zip,
       utility_id: id,
       predominant: row.Predominant === 'Yes' ? 1 : 0,
     });
   }
 
-  const utilitiesOut = stringify({
-    header: true,
-    columns: ['utility_id', 'name'],
-  });
-  utilitiesOut.pipe(
-    fs.createWriteStream(path.join(__dirname, 'data/utilities.csv')),
+  // Update authorities.json with the utilities from the dataset.
+  const authoritiesJsonPath = path.join(__dirname, '../data/authorities.json');
+  const authoritiesJson: AuthoritiesByState = JSON.parse(
+    await promisify(fs.readFile)(authoritiesJsonPath, 'utf-8'),
```

Comment:
optional – use the promises version of fs instead of promisify. Same with writeFile.